tracer/src/Datadog.Trace/Activity/ActivityListener.cs

```diff
@@ -286,5 +309,43 @@ private static void CreateDiagnosticSourceListenerInstance(Type diagnosticListen
 
             return typeBuilder.CreateTypeInfo()?.AsType();
         }
+
+        private static void CreateActivityKindSetter(Type activityType, Type activityKindType)
+        {
+            // Create dynamic method
+            // Input:
+            // [0] object (which we want to cast to System.Diagnostics.Activity type)
+            // [1] int (which we want to cast to System.Diagnostics.ActivityKind type)
+            // Steps:
+            // - Load the first argument (object)
+            // - Cast this value to System.Diagnostics.Activity type
+            // - Load the second argument (int)
+            // - Cast this value to System.Diagnostics.ActivityKind type
+            // - Call the Setter method for System.Diagnostics.ActivityKind.Set(System.Diagnostics.Activity, System.Diagnostics.ActivityKind)
+            // - Ret
+
+            // Fields: on an object
+            // Properties: Kinda like fields, they give you a _method_ under the hood for set/get
+            // activityType.GetProperty("Kind")!.SetMethod => actual (private) method that sets the value for the Kind property
+
+            // Get the actual runtime types from the System.Diagnostics.DiagnosticSource assembly
+            var runtimeActivityType = Type.GetType("System.Diagnostics.Activity, System.Diagnostics.DiagnosticSource", throwOnError: false) ?? activityType;
```

Comment:
nit: You already have the types since they're the arguments to this method. You should be able to use those rather than call `Type.GetType` again